dynamic_programming/hamiltons-cycle.cpp

```diff
@@ -0,0 +1,182 @@
+/**
+ * @author vakhokoto
+ * 
+ * The implementation of hamilton's cycle dynamic solution for 
+ * vertices number less than 20.
+*/
+#include <iostream>
+#include <vector>
+#include <cassert>
+
+typedef bool (*test_func_t)(void);
+
+/**
+ * The function determines if there is a hamilton's cycle in the graph
+ * 
+ * @param n is the number of vertices in the graph
+ * @param routes nxn boolean matrix of [i, j] where [i, j] is true if there is a road between from i to j
+*/
```

Comment:
```suggestion
 * @return `true` if ....
 * @return `false` if ...
 */
```
we use [doxygen](https://www.doxygen.nl/manual/docblocks.html) commenting style to auto-generate the documentation website - https://thealgorithms.github.io/c-plus-plus